src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2352,51 +2409,103 @@ private void ReportNullabilityMismatchWithTargetDelegate(SyntaxNode syntax, Name
             }
         }
 
-        private static TypeSymbolWithAnnotations InferResultNullability(BoundExpression operand, Conversion conversion, TypeSymbol targetType, TypeSymbolWithAnnotations operandType, bool fromConversionNode = false)
+        /// <summary>
+        /// Re-calculate and apply the conversion to the type of the operand and return the resulting type.
+        /// </summary>
+        private TypeSymbolWithAnnotations ApplyConversion(BoundExpression operand, Conversion conversion, TypeSymbol targetType, TypeSymbolWithAnnotations operandType)
         {
+            Debug.Assert(operand != null);
+            return ApplyConversion(operand, operand, conversion, targetType, operandType, checkConversion: true, fromExplicitCast: false, out _);
+        }
+
+        /// <summary>
+        /// Apply the conversion to the type of the operand and return the resulting type.
```

Comment:
>Apply the conversion to the type of the operand [](start = 12, length = 47)

- "Apply the conversion to the type (or expression) of the operand ..."?
- What is `node` for?
- typo: re-calculated